cloud/aws/resources/configs/misp/misp-init.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+#
+# This script performs post-setup customizations of a MISP instance.
+# It requires the presence of several environment variables being set,
+# typically passed in as part of the container environment.
+#
+# The exit codes signal different conditions:
+#   0 = initialization took place succesfully
+#   1 = an error occurred during initialization
+
+import os
+import socket
+import subprocess
+import sys
+import time
+import urllib3
+import yaml
+
+from pymisp import PyMISP
+
+CAKE = "/var/www/MISP/app/Console/cake"
+
+# Wait until the web server is up an running.
+print("waiting until MISP web server becomes available")
+while True:
+    try:
+        socket.socket().connect(("127.0.0.1", 443))
+        break
+    except Exception as e:
+        time.sleep(5)
+
+# When MISP runs for the first time, an admin user doesn't exist yet. Browsing
+# to the frontend triggers creation of an admin user lazily. But want eager
+# behavior on startup, so we must trigger 'cake UserInit' to perform the
+# intilization. The output includes an ephemeral API key that we can use
+# subsequently to connect via PyMISP.
+result = subprocess.run([CAKE, "UserInit"], capture_output=True)
+api_key = result.stdout.decode("utf-8").splitlines()[-1]
+if "MISP instance already initialised" in api_key:
+    print("skipping initialization of already existing admin user")
```

Comment:
This comment concerns all `print` statements.

Wouldn't it be better to integrate a logging library with proper configuration that defaults to logging to console? Proper logging has following advantages:

1. Ability to provide where (which line) code is broken
2. Log can be redirected to console, file, socket easily
3. Based on severity, logs can be suppressed
4. Maybe the most important thing, format can be changed in a central manner. (encapsulating log message in JSON for example)

Reply:
I agree with you for larger systems, but this is just a tiny independent startup script that works around the fact that these settings are not accessible through the MISP config files directly.